chem_evol.py

```diff
@@ -1894,14 +1894,25 @@ def __sn1a_contribution(self, i):
         self._imf(0, 0, -1, 0)
 
         # Get SN Ia yields
-        tables_Z = self.ytables_1a.metallicities
-        for tz in tables_Z:
-            if self.zmetal > tz:
-                yields1a = self.ytables_1a.get(Z=tz, quantity='Yields')
-                break
-            if self.zmetal <= tables_Z[-1]:
-                yields1a = self.ytables_1a.get(Z=tables_Z[-1], quantity='Yields')
-                break
+	#check available metallicities
+	tables_Z = self.ytables_1a.metallicities
+	#sort lowest to highest
+	tables_Z.sort(reverse=True)
+	for tz in tables_Z:
+	    # if current Z above available
+	    if self.zmetal > tz:
+		yields1a = \
+		self.ytables_1a.get(Z=tz, quantity='Yields')
+		break
+	    #if current Z below available
+	    if self.zmetal < tables_Z[-1]:
+		yields1a = \
+		self.ytables_1a.get(Z=tables_Z[-1],quantity='Yields')
+		break
+	    else:
+		if self.zmetal>=tz:
+			yields_extra = \
+			self.ytables_extra.get(Z=tz,quantity='Yields')
 
```

Comment:
If I understand, the two first "if" statements (line 1903 and 1908) are in case zmetal is higher or lower than the metallicity range covered by SNe Ia?  If so, you could put the "for" loop right after the "else" (line 1912).  But in that case, you would need to replace tz (line 1903) by tables_Z[0].